src/main/java/fizzbuzz/FizzBuzz.java

```diff
@@ -3,16 +3,91 @@
 
 package fizzbuzz;
 
+import java.util.Arrays;
+import java.util.List;
+
 public class FizzBuzz {
 
+	//openClosed verletzt -> nicht erweiterbar für extension
+
 	public String translate(int number) {
-		if (((number % 5) == 0) && ((number % 7) == 0)) // A multiple of both?
-			return "FizzBuzz";
-		else if ((number % 5) == 0)
-			return "Fizz"; // else a multiple of 5?
-		else if ((number % 7) == 0)
-			return "Buzz"; // else a multiple of 7?
-		else
-			return String.valueOf(number); // else just print it
+
+		String result = "";
+		boolean needsDefault = true;
+
+		var fooValidator = new FooValidator();
+		var fizzValidator = new FizzValidator();
+		var barValidator = new BarValidator();
+		var buzzValidator = new BuzzValidator();
+		
+		List<Validator> validators = Arrays.asList(fooValidator,fizzValidator,barValidator,buzzValidator );		
+
+		for (Validator validator : validators) {
+			if (validator.validate(number)){
+				result += validator.getTranslation();				
+				needsDefault = false;
+			}
+		}
+
+		return needsDefault ? String.valueOf(number) : result;

```

Comment:
This could also be done by checking the result instead of using a seperate boolean variable. If the result is "" than you can return the number itself.